esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -3155,31 +3158,78 @@ bool CFileSprayEx::onDropZoneFiles(IEspContext &context, IEspDropZoneFilesReques
         if (!context.validateFeatureAccess(FILE_SPRAY_URL, SecAccess_Read, false))
             throw MakeStringException(ECLWATCH_FILE_SPRAY_ACCESS_DENIED, "Permission denied.");
 
-        bool isLinux = false;
+        IpAddress ipToMatch;
+        const char* netAddress = req.getNetAddress();
+        if (!isEmptyString(netAddress))
+        {
+            ipToMatch.ipset(netAddress);
+            if (ipToMatch.isNull())
+                throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid server %s specified.", netAddress);
+        }
+
+        bool filesFromALinux = false;
         IArrayOf<IEspDropZone> dropZoneList;
         bool ECLWatchVisibleOnly = req.getECLWatchVisibleOnly();
-        IArrayOf<IConstTpDropZone> dropZones;
-        queryDropZoneInfo(context.getClientVersion(), nullptr, ECLWatchVisibleOnly, dropZones);
-        ForEachItemIn(currentDZIndex, dropZones)
+        Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory();
+        Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
+        Owned<IConstDropZoneInfoIterator> dropZoneItr = constEnv->getDropZoneIterator();
+        ForEach(*dropZoneItr)
         {
-            IConstTpDropZone& dz = dropZones.item(currentDZIndex);
-            const char* dzName = dz.getName();
+            IConstDropZoneInfo& dropZoneInfo = dropZoneItr->query();
+            if (ECLWatchVisibleOnly && !dropZoneInfo.isECLWatchVisible())
+                continue;
+
+            SCMStringBuffer dropZoneName, directory, computerName;
+            dropZoneInfo.getName(dropZoneName);
+            dropZoneInfo.getDirectory(directory);
+            dropZoneInfo.getComputerName(computerName); //legacy env
+            if (!dropZoneName.length() || !directory.length())
+                continue;
 
-            IArrayOf<IConstTpMachine>& servers = dz.getTpMachines();
-            ForEachItemIn(j, servers)
+            bool isLinux = false;
+            if (computerName.length() > 0)
+            { //legacy env
+                StringBuffer dir = directory.str();
+                Owned<IConstMachineInfo> machine = constEnv->getMachine(computerName.str());
+                if (machine->getOS() == MachineOsLinux || machine->getOS() == MachineOsSolaris)
+                {
+                    dir.replace('\\', '/');//replace all '\\' by '/'
+                }
+                else
+                {
+                    dir.replace('/', '\\');
+                    dir.replace('$', ':');
+                }
+                directory.set(dir.str());
+            }
```

Comment:
No special handling of legacy computerName dropzones should be here, backward compatibility is handled by getDropZoneIterator().

I am still not convinced by the translation, but why does it apply to legacy anymore than new?